gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -234,6 +236,48 @@ private boolean isDirectPathEnabled(String serviceAddress) {
     return false;
   }
 
+  // DirectPath should only be used on Compute Engine. Follow this doc:
+  // https://docs.google.com/document/d/1xQXE27x9wTvwPsgiX9Hn0o8mcq5z3SKi-1jwscQsCAk/edit#.
+  private boolean isOnComputeEngine() {
+    String osName = System.getProperty("os.name");
+    Process process;
+    String result = "";
+    String tmp = null;
+    if ("Linux".equals(osName)) {
+      String cmd = "cat /sys/class/dmi/id/bios_date && cat /sys/class/dmi/id/bios_vendor";
+      try {
+        process = Runtime.getRuntime().exec(new String[] {"/bin/sh", "-c", cmd});
+        process.waitFor();
+        BufferedReader br = new BufferedReader(new InputStreamReader(process.getInputStream()));
+        while ((tmp = br.readLine()) != null) {
+          result = result + tmp + ",";
+        }
+        return result.contains("01/01/2011") && result.contains("Google");
+      } catch (IOException e) {
+        e.printStackTrace();
```

Comment:
Printing a stack trace isn't helpful. Either return a correct value if possible, or throw/bubble an exception if not.

Reply:
Directly return false when exception is catched.